solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -1278,6 +1287,15 @@ public Builder withMaxConnectionsPerHost(int max) {
       return this;
     }
 
+    /**
+     * Specify the maximum number of concurrent asynchronous requests the client can send. The
+     * default value is 1000
+     */
+    public Builder withMaxOutstandingRequests(int max) {
```

Comment:
If I'm not mistaken it's actually a maximum amount per host. Could we make it explicit, e.g., `maxOutstandingRequestsPerHost`?

Reply:
well I don't think this is per-host. The Async tracker doesn't do any per-host logic, does it?

Reply:
Yeah, it's really 2 things:
1. throttles maximum number of outstanding requests across all hosts
2. the number of requests per host that will be buffered waiting to send, once 1. hits its limit.

2. does buffering, but no actual throttling, and IMO it's not useful. If we proceed with making the "per host" part hardcoded (removing 2.), then the javadocs here for `withMaxOutstandingRequests(int)` will be completely accurate. And as it stands I'd say they're essentially accurate, practically speaking, since the per-host logic that they do influence is more of an internal detail.